### PR TITLE
feat: SendUpdates() API to send only extension data to via existing requests

### PR DESCRIPTION
**Pull request by rvagg**
Closes: https://github.com/ipfs/go-graphsync/issues/348

WIP, figuring out the testing parts of this currently

### CODE REVIEW

File: impl/graphsync_test.go
Comment (rvagg) [inline impl/graphsync_test.go:1822]: @hannahhoward I need a bit of help with this test. I'm doing a two-way "update" message, checking that I can send and receive extensions both ways. I've got it working up until here, where the test fails, "response channel never closed". I don't think the unpause is working properly, something about the way I've messed up the response stream perhaps. The logic involved in moving these things around is stretching my comprehension of the internals so there's a good chance I'm not even doing this right at the top level (e.g. is my use of the hooks to receive the extension data the right thing to be doing with these messages?).

Comment (rvagg) [reply impl/graphsync_test.go:1826]: After I wrote this I had a brainwave that it might have been about my use of hooks + channels, and it turns out that was correct. I was blocking in the test, not inside GS. This passes now.

**Comment by rvagg** [comment]
Relevant race failure in [one of the actions runners](https://github.com/ipfs/go-graphsync/runs/5197383756?check_suite_focus=true):

```
  === RUN   TestUpdateResponse/while_unpaused
      responsemanager_test.go:901: 
          	Error Trace:	channelassertions.go:44
          	            				channelassertions.go:15
          	            				responsemanager_test.go:901
          	Error:      	Not equal: 
          	            	expected: 0
          	            	actual  : 1
          	Test:       	TestUpdateResponse/while_unpaused
          	Messages:   	should send first extension response
  === RUN   TestUpdateResponse/while_paused
  --- FAIL: TestUpdateResponse (10.12s)
      --- FAIL: TestUpdateResponse/while_unpaused (10.00s)
      --- PASS: TestUpdateResponse/while_paused (0.12s)
```

I'll have to figure that one out before this is merged.

**Comment by rvagg** [comment]
Can't repro the error locally with or without race, on 1.17 or 1.16. I've moved around a `synchronize()` call but I don't think that's a good explanation for the failure. It remains a bit of a concern since I can't explain it but I'll wait for review because maybe I'm missing something obvious.

**Review by hannahhoward — APPROVED** [review]
LGTM thought my brain is too broke to decipher the cause of your race. will look in morning.

**Comment by rvagg** [comment]
https://github.com/ipfs/go-graphsync/pull/360 fixed the race; so this is now squashed, rebased and ready to roll.

**Pull request merged**
